File: Java/IntegerListFromUsp/src/Exercicio04.java
import java.util.Scanner;

public class Exercicio04 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.reset();
        System.out.print("Digite um número: ");
        int numero = scanner.nextInt();
        System.out.print("Digite um expoente: ");
        scanner.reset();
        int exp = scanner.nextInt();
        if (exp > 0) {
            int resultado = (int) Math.pow((double) numero, (double) exp);
            System.out.println("Resultado %d".formatted(resultado));
        } else {
            System.out.println("Expoente não pode ser negativo.");
        }
    }
}

File: CPP/IntegerListFromUsp/src/Exercicio04.cpp
#include <iostream>

using namespace std;

int main() {
    int x, n;
    cout << "Digite x: ";
    cin >> x;
    cout << "Digite n (não-negativo): ";
    cin >> n;

    if (n < 0) {
        cout << "n deve ser não-negativo." << endl;
        return 1;
    }

    long long resultado = 1;
    for (int i = 0; i < n; i++) {
        resultado *= x;
    }
    cout << x << "^" << n << " = " << resultado << endl;
    return 0;
}
